hedera-mirror-rest/transactions.js

```diff
@@ -786,18 +784,70 @@ const extractSqlFromTransactionsByIdOrHashRequest = async (transactionIdOrHash,
 
   if (nonce !== undefined) {
     params.push(nonce);
-    mainConditions.push(`${Transaction.NONCE} = $${params.length}`);
+    conditions.push(`${Transaction.NONCE} = $${params.length}`);
   }
 
   if (scheduled !== undefined) {
     params.push(scheduled);
-    mainConditions.push(`${Transaction.SCHEDULED} = $${params.length}`);
+    conditions.push(`${Transaction.SCHEDULED} = $${params.length}`);
+  }
+
+  return {conditions, scheduled, nonce};
+};
+
+const getTransactionsByTransactionIdsSql = (transactionKeys, filters, timestampField) => {
+  let payerAccountParams = [];
+  let params = [];
+
+  let minTimestamp = 0;
+  let idConditions = [];
+  let includePayer = false;
+
+  if (transactionKeys.length % 2 !== 0) {
+    throw new InvalidArgumentError('transaction keys must be in sequenced pairs of [payer,timestamp,payer,timestamp]');
+  }
+
+  for (let index = 0; index < transactionKeys.length; index += 2) {
+    let paramIndex = params.length;
+    const payer = transactionKeys[index];
+    const timestamp = BigInt(transactionKeys[index + 1]);
+
+    if (payer) {
+      payerAccountParams.push(`$${++paramIndex}`);
+      idConditions.push(`(${Transaction.PAYER_ACCOUNT_ID} = $${paramIndex} and 
+                               ${timestampField} = $${++paramIndex})`);
+      params.push(payer);
+      includePayer = transactionKeys;
+    } else {
+      idConditions.push(`${timestampField} = $${++paramIndex}`);
+    }
+    params.push(timestamp);
+
+    if (!minTimestamp || timestamp < minTimestamp) {
+      minTimestamp = timestamp;
+    }
   }
-  mainConditions.unshift(...commonConditions);
+  const maxConsensusTimestamp =
+    timestampField === Transaction.CONSENSUS_TIMESTAMP
+      ? transactionKeys[transactionKeys.length - 1]
+      : minTimestamp + maxTransactionConsensusTimestampRangeNs;
```

Comment:
we need to track `maxTimestamp` and change this to
```suggestion
      : maxTimestamp + maxTransactionConsensusTimestampRangeNs;
```

Reply:
do we though? For a batch, the batch is going to have some validstartns so all inner transactions will have to execute with in the batch.validstart + (35 min or whatever the max is).

Reply:
I don't think so. The only relevant description in the HIP I can find is:

```
Inner Transactions, by design:
...
4. Are deduplicated on an individual basis. If two inner transactions within the same batch have the same TransactionID, the transaction will be rejected.
```

If they all have the same valid start timestamp, with the rule above, a payer can't have more than 1 inner transaction in the batch, since otherwise multiple inner transactions end up with the same transaction id.

Reply:
I see what you mean by we can just use batch.validstartns + a max range.. btw, why can' we just store payer + consensus timestamp of all inner transactions?

Reply:
We use valid start so that we can capture any child/preceding transactions for an inner whenever we are querying for things. We could use consensusTimestamp but would need to store more data in the array in the db.